inkfish/cmds.py

```diff
@@ -34,29 +40,51 @@ def pot(args=sys.argv):
     parser = create_pot_parser()
     args = parser.parse_args(args=args[1:])
 
-    discriminant = create_discriminant(args.discriminant_challenge)
+    discriminant = create_discriminant(args.discriminant_challenge, args.length)
     if args.verbose:
+        print("proof type: %s" % args.type)
         print("discriminant: %s" % discriminant)
+        print("discriminant size: %s" % args.length)
 
     generator = ClassGroup.from_ab_discriminant(2, 1, discriminant)
     if args.verbose:
         print("generator: %s" % str(generator))
 
     if args.proof:
-        ok = check_proof_of_time(
-            discriminant, generator, args.proof, args.iterations, args.length)
+        if args.type == "wesolowski":
+            ok = check_proof_of_time_wesolowski(
+                discriminant, generator, args.proof, args.iterations, args.length)
+        elif args.type == "n-wesolowski":
+            ok = check_proof_of_time_nwesolowski(
+                discriminant, generator, args.proof, args.iterations, args.length)
+        elif args.type == "pietrzak":
+            ok = check_proof_of_time_pietrzak(
+                discriminant, generator, args.proof, args.iterations, args.length)
+        else:
```

Comment:
You can remove this "else" if you use "choices" (which will also print the choices in the `-h` help).